LBHackney-IT/lbh-fss-portal-frontend
# Feature/manage taxonomies

**Description**

Allow admin users to manage service categories and service demographics on a new 'Taxonomies' page. These taxonomies govern the questions which appear on the 'What you do' and 'Who you work with' pages of the service form.

**Test instructions**

- pull this branch 
- run application with `yarn dev` 
- navigate to **/taxonomies**
- check you can add/update/remove taxonomies, you should receive success messages (any changes will not take effect as mock data is used)
- check failed responses from the api are handled, do this by:
    - opening the /mock-api/taxonomies/POST.js file and replace the 200 status with the 401 status, then try adding a taxonomy term 
    - opening the /mock-api/taxonomies/__ termId __/PATCH.js  file and replace the 200 status with the 401 status, then try editing a taxonomy term
    - opening the /mock-api/taxonomies/DELETE.js file and replace the 200 status with the 401 status, then try deleting a taxonomy term 
- check non-admin users are unable to reach the /taxonomies page, do this by: 
    - switching to a non-admin user by commenting out the ADMIN section in /mock-api/account and uncommenting lines 34-39 (the taxonomies tab should be removed from the menu and if trying to navigate to **/taxonomies** directly you should get an Access Denied message) 
- check changes in the taxonomy data result in changes to the service form, do this by
    - opening the /mock-api/mockTaxonomies.json file and removing some of the categories and demographics
    - restart application (turn project off, then `yarn dev` again) 
    - add a new service under the 'Your listings' tab 
    - check the deleted  categories and demographics do not appear on the 'What you do' and 'Who you work with' pages 

any questions let me know 

**Ticket**

https://app.clickup.com/t/a6zw30
